### PR TITLE
Added S3Files field.

### PR DESCRIPTION
**Pull request by jeffreypriebe**
Similar to singular S3File field, this patch uses the multiple LocalFiles field and provides S3 functionality for multiple file uploads.

S3Files-field

### CODE REVIEW

**Comment by victorkurauchi** [comment]
@jeffreypriebe i was implementing s3 multiple files yesterday... and found yours. haha thank you, helped a lot.

File: fields/types/s3files/S3FilesType.js [hunk 1]
```diff
@@ -0,0 +1,572 @@
+/*!
+ * Module dependencies.
+ */
+
+var _ = require('underscore'),
+	moment = require('moment'),
+	keystone = require('../../../'),
+	util = require('util'),
+	knox = require('knox'),
+	// s3 = require('s3'),
```
Comment (ericelliott) [inline fields/types/s3files/S3FilesType.js]: Is this needed, or not?

Comment (jeffreypriebe) [inline fields/types/s3files/S3FilesType.js]: The "s3" require? Guess not. [Knox](https://github.com/Automattic/knox) does the S3 work.

**Comment by jeffreypriebe** [comment]
@victorkurauchi Good to hear. Anything to tweak/add/fix?

**Comment by morenoh149** [comment]
@jeffreypriebe removing the commented s3 line would be best

**Comment by riyadhalnur** [comment]
there's already a PR open for multiple S3 files #1362

**Comment by JedWatson** [comment]
Thanks for the work here @jeffreypriebe and sorry it didn't get merged. We're going to take a different approach for file fields with pluggable storage post 0.4, and need to cut down the surface area to get that out first.

**Comment by jeffreypriebe** [comment]
No worries - lots of good stuff happening on 0.4 and excited to see that coming through. Will have more work to do to bring some of my work into 0.4 once that settles down.

Pluggable storage, custom data types without hacking the source - all good things coming in 0.4.

**Pull request closed without merging**